torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -1906,34 +1890,46 @@ Tensor kl_div_target_backward(
     bool log_target) {
   Tensor grad_target;
   if (!log_target) {
-    if (!areAnyTensorSubclassLike({self, target}) &&
-        !grad_output._is_zerotensor()) {
-      grad_target = grad_output.mul(target.log().add_(1).sub_(self))
-                        .masked_fill_(target == 0, 0.);
-    } else {
-      grad_target = grad_output.mul(target.log().add(1).sub(self))
-                        .masked_fill(target == 0, 0.);
-    }
+    grad_target = grad_output.mul(target.log().add(1).sub(self))
+                      .masked_fill(target == 0, 0.);
   } else {
-    if (!areAnyTensorSubclassLike({self, target})) {
-      grad_target =
-          grad_output.mul(target.add(1).sub_(self).mul_(target.exp()));
```

Comment:
Making this `sub_` alone out-of-place seemed to fix it. And leaving this sub in-place, and making the rest of them out-of-place still errors.